MainModule/Server/Plugins/Server-SoftShutdown.lua

```diff
@@ -61,14 +62,15 @@ return function(Vargs, GetEnv)
 
 				Logs:AddLog("Script", `Teleporting {player.Name} back to the main game`)
 				teleportedPlayers[player] = 1
-				TeleportService:Teleport(game.PlaceId, player, {[PARAMETER_2_NAME] = true})
+				table.insert(playersToTeleport, player)
 			end
 		end
 
 		service.Events.PlayerAdded:Connect(teleport)
 		for _, player in ipairs(service.GetPlayers()) do
 			teleport(player)
 		end
+		TeleportService:TeleportPartyAsync(game.PlaceId, playersToTeleport, {[PARAMETER_2_NAME] = true})
```

Comment:
This only works for current players you still have to support `service.Events.PlayerAdded:Connect(teleport)`